Suggested changes to this template

colleenXu opened the pull request:
@karafecho 

I have two suggestions/questions for this template:
* perhaps the CQS should use [Service-Provider-TRAPI](http://smart-api.info/registry?q=36f82f05705c317bac17ddae3a0ea2f0) which is a KP, rather than BTE which is an ARA? Service-Provider-TRAPI will also return faster (working off of a smaller set of underlying KPs). However, it won't score the results returned. 
* the attribute-constraint value should be an int (number) rather than a string? A string doesn't seem to make sense with the ">" operator 

And @mbrush: I don't have an objection to using "biolink:evidence_count" for the attribute-type-id. But I am confused on why we use it over ["biolink:has_count"](https://github.com/biolink/biolink-model/blob/dc87efcf374cd80a462d62a9087978bee1e73a4c/biolink-model.yaml#L817) or ["biolink:concept_pair_count"](https://github.com/biolink/biolink-model/blob/dc87efcf374cd80a462d62a9087978bee1e73a4c/biolink-model.yaml#L5711).

colleenXu committed:
mvp1-template4-bte-aeolus: change tool to service-provider-trapi

colleenXu committed:
mvp1-template4-bte-aeolus: use int rather than string for attribute-constraint

karafecho reviewed (commented):
Your suggested changes make sense to me. The lack of scoring with infores:service-provider-trapi" won't be an issue. Please change the file name to "mvp1-template5-service-provider-aeolus".

I'll let @mbrush respond to your query re "biolink:evidence_count" before approving the PR.

mbrush commented:
1. re: 'Service-Provider-TRAPI' as the as the executor of the 'lookup' operation in a CQS-specified workflow  - my understanding was that an ARA should be specified here, based on the other CQS templates I have seen.  And if there is a specific KP that should be queried by that ARA to answer the query, this is specified in the query block, using an `allow_list`  under a `provided_by` object.  This is how it works in the [clinical-kps mvp1 template](https://github.com/TranslatorSRI/CQS/blob/ccf2a9ab17234329658133d867828dd3473af6d0/templates/mvp1-templates/mvp1-template1-clinical-kps/mvp1-template1-clinical-kps.json) at least.   @karafecho can you confirm rules/conventions about where to specify specific ARAs vs KPs in allow lists in the different blocks of a CQS template?  If it is ok to put a KP in a lookup operation allow list, and this results in faster performance for this template - perhaps that would be best as Colleen suggests?  In this case, scoring would be performed by whichever of the several existing scoring services returns a score first, right?
 
2. re: the evidence count value being a string - agreed this should be an int. 
 
3. finally, I agree that "has_count" would be most appropriate, but it is designated as a node property in Biolink. @sierra-moxon  perhaps we can look into making this an edge property? I can’t imagine a use case for it being used as a node property.

maximusunc commented:
Using `infores:service-provider-trapi` is a valid executor of the 'lookup' operation. This does not have to be an ARA, it just has to be a service registered on SmartAPI and that broadcasts that it can perform said operation.

On a related note, like Colleen pointed out, a "score" operation would then need to be added since service-provider-trapi does not do scoring internally like biothings-explorer does.

colleenXu commented:
So the conversation so far sounds like:

* we all agree on changing the constraint value type from string `"20"` to int `20`
* it's unclear if we want to change the infores to `service-provider-trapi`. If we made the change, there'd need to be an additional operation where another service scores the results. 
    * It sounds like it may be simpler to keep using `biothings-explorer` and getting scored results back....so I'm okay reverting this. What do you think, @maximusunc and @karafecho?
* It's unclear if we want to change the attribute-type-id. I think it's fine to go forward with `evidence_count`, rather than wait for a biolink-model change. What do you think, @mbrush?

Once it's clear how to proceed, I'll adjust this PR...

sierra-moxon commented:
`biolink:has_count` is currently mapped to `LOINC:has_count` -- I signed up for an account and will try poking around to find where this mapping might have come from.  It would be good to figure that out before changing it to an edge property, unless one of you already know why this was added and by whom.

`biolink:concept_pair_count` and `concept count object` have very similar definitions, are both edge properties, and are both integers.  It would be great to understand how to appropriately apply these or if they were added for a specific group or purpose.

`biolink:evidence_count`: is an edge property and an **integer** in Biolink.

karafecho commented:
FWIW, adding a "score" operation amounts to nothing other than this:

```
{
    "workflow": [
        {
            "id": "lookup",
            "runner_parameters": {
                "allowlist": ["infores:service-provider-trapi"]
            }
        },
        {
            "id":"score"
        }
```
So, I think you need to weigh the pros and cons of the following two options: (1) sticking with `infores:biothings-explorer` despite the inefficiencies you've noted and using their scoring (perhaps because it's more appropriate for this template?) or (2) switching to `infores:service-provider-trapi` for the reasons you've outlined but relying on other ARAs for the scoring.

colleenXu committed:
Update and rename mvp1-template4-bte-aeolus.json to mvp1-template5-service-provider-aeolus.json

Following Kara's suggestions in https://github.com/TranslatorSRI/CQS/pull/9#issuecomment-2013754633

colleenXu commented:
@karafecho Given your feedback, I renamed the file and changed the workflow to what you provided. Let's try this out?

karafecho reviewed (approved):
Looks good to me, but I'll let @maximusunc and @jdr0887 provide the final approval.

maximusunc reviewed (approved):
This looks good to me!
